### ISSUE TITLE
Ignore comment annotation doesn't work for one-line functions

### ISSUE THREAD

**Issue by yang** [pwlmaciejewski/coffeelint-variable-scope#7]
```
$ cat ./go.coffee
'use strict'
x = 0
###coffeelint-variable-scope-ignore###
-> x = 1

$ ./node_modules/.bin/coffeelint ./go.coffee
  ✗ ./go.coffee
     ✗ #2-4: Outer scope variable overwrite. x.

✗ Lint! » 1 error and 0 warnings in 1 file
```

**Comment by AsaAyers** [comment]
CoffeeLint comes with a mechanism to do this. I'm not sure why this rule is providing a custom mechanism that no other rule provides. The method below doesn't rely on scopes, it disables between any two lines.

```
'use strict'
x = 0
# coffeelint: disable=variable_scope
-> x = 1
# coffeelint: enable=variable_scope
```

**Comment by yang** [comment]
I had tried that before but it sadly doesn't work for this rule:

```
$ cat go.coffee
'use strict'
x = 0
# coffeelint: disable=variable_scope
-> x = 1
# coffeelint: enable=variable_scope
->
  # coffeelint: disable=variable_scope
  x = 1
  # coffeelint: enable=variable_scope

$ ./node_modules/.bin/coffeelint ./go.coffee
  ✗ ./go.coffee
     ✗ #2-4: Outer scope variable overwrite. x.
     ✗ #2-8: Outer scope variable overwrite. x.

✗ Lint! » 2 errors and 0 warnings in 1 file
```

**Comment by AsaAyers** [comment]
oh, this rule isn't warning you around where the error is produced (line 4) above. It's wrapping the error all the way from where the variable was defined. This seems awkward but to use the built in capability to disable a rule you need to use:

```
$ cat go.coffee
'use strict'
# coffeelint: disable=variable_scope
x = 0 # Now this will suppress all errors that start on this line.
# coffeelint: enable=variable_scope
-> x = 1
->
  x = 1
```

CoffeeLint filters based on the first line, which is known for all errors/warnings. [implementation](https://github.com/clutchski/coffeelint/blob/master/src/coffeelint.coffee#L293-L315)

I have no control over this project, I'm the maintainer of CoffeeLint. I Sometimes subscribe to rules I don't use just to see what others are doing with CoffeeLint.

**Comment by yang** [comment]
Got it.  I believe that, usability-wise, the most common use case is to
selectively control whether a particular mutation is kosher (rather than
blanketing a variable altogether).  But anyway this is an adequate
workaround for now for my particular code base, thank you!
